src/plonk/prover.rs

```diff
@@ -184,17 +233,38 @@ impl<C: CurveAffine> Proof<C> {
             let mut deltaomega = C::Scalar::one();
             for &(wire, _) in wires.iter() {
                 let omega = domain.get_omega();
-                parallelize(&mut modified_advice, |modified_advice, start| {
-                    let mut deltaomega = deltaomega * &omega.pow_vartime(&[start as u64, 0, 0, 0]);
-                    for (modified_advice, advice_value) in modified_advice
-                        .iter_mut()
-                        .zip(witness.advice[wire.0][start..].iter())
-                    {
-                        // Multiply by p_j(\omega^i) + \delta^j \omega^i \beta
-                        *modified_advice *= &(deltaomega * &x_0 + &x_1 + advice_value);
-                        deltaomega *= &omega;
+                match wire {
+                    Wire::Advice(wire) => {
+                        parallelize(&mut modified_advice, |modified_advice, start| {
+                            let mut deltaomega =
+                                deltaomega * &omega.pow_vartime(&[start as u64, 0, 0, 0]);
+                            for (modified_advice, advice_value) in modified_advice
+                                .iter_mut()
+                                .zip(witness.advice[wire.0][start..].iter())
```

Comment:
Similarly, consolidate the cases together by putting the match at this level instead.